### ISSUE TITLE
cant display a folder with 2000 files

### ISSUE THREAD

**Issue by Piour** [cozy/cozy-files#223]
I did upload 2000 pictures from my phone to my cozy, and i cant open the folder with 2000 files in cozy-files (timeout)

**Comment by jsilvestre** [comment]
Oh this had to happen one day...it's obviously not something intended, but we were fully aware that it would go wrong at some point.

**Comment by poupotte** [comment]
From onome in [forum](https://forum.cozy.io/t/messages-derreur-dans-lapplication-files/560)
He had diffuclties to display lots of files (about 1000). When he clicks on folder with 1000 files, files are displayed only after a long time and one or two errors as 'Script are busy or doesn't respond'

**Comment by m4dz** [comment]
This is definitely a sad feature :cry:. Maybe we should rethink the way our files views are displayed, I have some ideas, but not the time to tests and implement them. Keep an eye on it.

**Comment by frankrousseau** [comment]
I think we should try again to put a timeout between each row append for list of more than 50 files. It could be a good quick and dirty fix.

**Comment by jsilvestre** [comment]
Code has landed to development, so you should be able to test it soon. We're expecting your feedback to  keep improving it.

**Comment by frankrousseau** [comment]
It's better, it doesn' block the navigator too long. But it's still slow with 655 files. Do you think we can add pagination to it?

**Comment by frankrousseau** [comment]
Another issue is that keyboard navigation doesn't work, pageup, pagedown, up and down doesn't scroll the list.

**Comment by frankrousseau** [comment]
When progress barr is displayed, the white background doesn't cover the whole list. It's grey at the bottom.

**Comment by pierrerousseau** [comment]
Yeah! I can display my 2000+ files directory!

**Comment by jsilvestre** [comment]
* Pagination is one step further, but it's not trivial considering the cell-reuse component (https://github.com/cozy/cozy-files/issues/349).
* I'm opening an issue for the keyboard shortcuts thing (https://github.com/cozy/cozy-files/issues/347).
* I'm opening an issue for the style problem (https://github.com/cozy/cozy-files/issues/348).

**Comment by poupotte** [comment]
@frankrousseau , @jsilvestre Can we consider this issue fixed ?

**Comment by pierrerousseau** [comment]
for me yes, great job!
